File: webapp/src/schemas/job-posting.ts
import { z } from 'zod';

export const jobPostingSchema = z.object({
  id: z.string(),
  vendor: z.string(),
  company: z.string(),
  title: z.string(),
  team: z.string().nullable(),
  location: z.string().nullable(),
  commitment: z.string().nullable(),
  hosted_url: z.string(),
  apply_url: z.string().nullable(),
  posted_at: z.string().nullable(),
  is_active: z.boolean(),
});

export const jobPostingListSchema = z.object({
  items: z.array(jobPostingSchema),
  total: z.number(),
  page: z.number(),
  page_size: z.number(),
});

export const quickApplyResponseSchema = z.object({
  status: z.string(),
});

export type JobPosting = z.infer<typeof jobPostingSchema>;
export type JobPostingList = z.infer<typeof jobPostingListSchema>;
export type QuickApplyResponse = z.infer<typeof quickApplyResponseSchema>;

export const QUICK_APPLY_VENDORS = new Set(["lever"]);

File: backend/app/models/job_posting.py
from __future__ import annotations

from datetime import datetime

from app.models.common import ApiModel


class JobPostingOutput(ApiModel):
    id: str
    vendor: str
    company: str
    title: str
    team: str | None
    location: str | None
    commitment: str | None
    hosted_url: str
    apply_url: str | None
    posted_at: datetime | None
    is_active: bool


class JobPostingListResponse(ApiModel):
    items: list[JobPostingOutput]
    total: int
    page: int
    page_size: int


class JobPostingSyncResponse(ApiModel):
    vendors_synced: list[str]
    vendors_failed: list[str]
    postings_seen: int
    postings_created: int
    postings_updated: int
    postings_deactivated: int


class QuickApplyRequest(ApiModel):
    profile_id: str
    phone: str | None = None
    comments: str | None = None


class QuickApplyResponse(ApiModel):
    status: str
